src/rgw/rgw_file.cc

```diff
@@ -987,9 +995,17 @@ namespace rgw {
       data.swap(orig_data);
 
       /* restart processing with different oid suffix */
-      dispose_processor(processor);
-      processor = select_processor(*static_cast<RGWObjectCtx *>(s->obj_ctx),
-				   &multipart);
+      processor = boost::none;
```

Comment:
is this really needed? does `boost::optional` try to call `RGWPutObjProcessor_Atomic::operator=()` otherwise?